src/status_im/contexts/keycard/change_pin/events.cljs

```diff
@@ -46,15 +46,18 @@
                                 (rf/dispatch [:navigate-back])
                                 (rf/dispatch [:keycard/disconnect])
                                 (rf/dispatch [:open-modal :screen/keycard.pin-change-failed]))
-                 :current-pin current-pin
-                 :new-pin     new-pin}]))
+                 :current-pin (security/safe-unmask-data (:current-pin change-pin-data))
+                 :new-pin     (security/safe-unmask-data (:new-pin change-pin-data))}]))
 
 (defn- verify-pin-and-continue
-  [current-pin new-pin]
+  [change-pin-data]
   (rf/dispatch
    [:keycard/verify-pin
-    {:pin        current-pin
-     :on-success #(change-pin-and-continue current-pin new-pin)
+    {:pin        (security/safe-unmask-data (:current-pin change-pin-data))
+     :on-success (fn []
+                   (rf/dispatch [:keycard/connect.next-stage
```

Comment:
Thank you @flexsurfer for adding this.

Should we use a different name for this event? We usually use verbs in event name part, like connect, update, get, change, dispatch etc. This one feels a little off. Maybe like update-next-stage etc.

Reply:
thanks @Parveshdhull ., so basically it's still connect , probably we could pass next-stage? as param, would be even better, but lets make this as separate task

Reply:
i can do it in this PR https://github.com/status-im/status-mobile/pull/22214/files, anyway it's better to wait till it merged, we don't have key-uid param anymore

Reply:
done, i've added next-stage? param im my PR